### PR TITLE
fix: update footer to not be fixed to bottom of viewport

### PR DESCRIPTION
**Pull request by tusharwebd**
Issue: #48

# Summary

This fixes issue #48

## Requirement:

The footer is fixed at bottom of viewport and overlaps the other elements. We want them to be at the bottom of the whole page, not fixed to the bottom of the viewport.

## Changes made:

Added tailwind classes to ensure main component takes 100% of viewport and used `flex` classes to convert component into flex containers.

### Screenshots / Videos 

https://github.com/amlan-roy/resume-craft/assets/59330872/2606e3c9-5ab4-4e1e-b563-a87738ab071f

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by tusharwebd** [comment]
I see unit tests failed. Let me check on this.

File: src/app/layout.tsx
Comment (amlan-roy) [inline src/app/layout.tsx:1]: This works. However, I would suggest using `cn()` function from "@/lib/utils" for consistency here.

You can use something like 
`className={cn(inter.className, "min-h-screen flex flex-col")}`

**Review by amlan-roy — CHANGES_REQUESTED** [review]
We would also need to add `className="flex-grow"` in all the `layout.tsx` files (except the`src\app\layout.tsx` file). Otherwise, the footer would not be pushed down if the height of the main content does not cover the full screen. You can refer the [deployment ](https://resume-craft-git-fork-tusharwebd-foo-1c7ec7-amlan-roys-projects.vercel.app/home)of this branch to see this behaviour.

File: src/app/layout.tsx
Comment (amlan-roy) [reply src/app/layout.tsx:1]: When making changes for this, I would also suggest updating the structure of the elements as detailed in the issue description. Thanks.

**Comment by tusharwebd** [comment]
> We would also need to add `className="flex-grow"` in all the `layout.tsx` files (except the`src\app\layout.tsx` file). Otherwise, the footer would not be pushed down if the height of the main content does not cover the full screen. You can refer the [deployment ](https://resume-craft-git-fork-tusharwebd-foo-1c7ec7-amlan-roys-projects.vercel.app/home)of this branch to see this behaviour.

Makes sense. Checking on this.

**Comment by tusharwebd** [comment]
@amlan-roy, I've requested access for the vercel deployment.

File: src/app/layout.tsx
Comment (amlan-roy) [reply src/app/layout.tsx:1]: Made the changes

**Review by amlan-roy — APPROVED** [review]
LGTM! 🚀
Made the remaining changes since it is required for the ongoing changes. Thanks for the efforts @tusharwebd.

**Pull request merged**